src/styled-template-language-service.ts

```diff
@@ -240,13 +241,22 @@ export default class StyledTemplateLanguageService implements TemplateLanguageSe
         const doc = this.createVirtualDocument(context);
         const virtualPosition = this.toVirtualDocPosition(position);
         const stylesheet = this.scssLanguageService.parseStylesheet(doc);
+        const emmetResults: vscode.CompletionList = {
+            isIncomplete: true,
+            items: []
+        }
+        this.cssLanguageService.setCompletionParticipants([getEmmetCompletionParticipants(doc, virtualPosition, 'css', {}, emmetResults)]);
         const completionsCss = this.cssLanguageService.doComplete(doc, virtualPosition, stylesheet) || emptyCompletionList;
         const completionsScss = this.scssLanguageService.doComplete(doc, virtualPosition, stylesheet) || emptyCompletionList;
         completionsScss.items = filterScssCompletionItems(completionsScss.items);
         const completions: vscode.CompletionList = {
             isIncomplete: false,
             items: [...completionsCss.items, ...completionsScss.items],
         };
+        if (emmetResults.items.length) {
+            completions.items.push(...emmetResults.items);
+            completions.isIncomplete = true;
```

Comment:
Just a heads up: `isIncomplete` is ignored by TS so setting it to true won't have any effect. Does emmet depend on this?

Reply:
yes, it does. For example: `m` -> `margin: ;` and `mt` -> `margin-top: ;`. So the completions need to be re-calculated for almost every char

Reply:
Opened https://github.com/Microsoft/TypeScript/issues/21999. I've discussed this problem with TS before but emmet may be the first real use case for it.

As we discussed offline, I believe we should be able to take in this PR even without `isIncomplete`. Users will just have to manually trigger completions in some cases. Do you agree @ramya-rao-a?

Reply:
Agreed